src/object/srv_ec_aggregate.c

```diff
@@ -35,15 +35,37 @@
  *	- The parity for peer parity extents is transferred.
  *	- Replicas for the stripe are removed from parity targets.
  *
+ * If replicas are partial, and prior parity exists:
+ *	- If less than half cells are updated (have replicas, parity is updated:
+ *		- Old data cells for cells with replica data are fetched from
+ *		  data targets (old, since fetched at epoch of existing parity).
+ *		- Peer parity is fetched.
+ *		- Parity is incrementally updated.
+ *		- Updated parity is transferred to peer parity target(s).
+ *	- If half or more of the cells are update by replicas:
+ *		- All cells not filled by local replicas are fetched.
+ *		- New parity is generated from entire stripe.
+ *		- Updated parity is transferred to peer parity target(s).
+ *	- Replicas for the stripe are removed from parity targets.
+ *
+ * If the stripe contains holes later than the parity:
+ *	- Valid ranges in the stripe are pulled from the data targets and
+ *	  written to local VOS, and peer parity VOS, as replicas.
+ *	- Parity is removed for latest parity epoch in local VOS,
+ *	  and from VOS on peer parity targets.
```

Comment:
L51 ~L55 is for the case that a lower epoch parity exists, and then at higher epoch there is some overlapped punch ext?

Reply:
Correct. The stripe incurred a punch to a partial range at an epoch later than the stripe's parity.